CIP-_/README.md

```diff
@@ -0,0 +1,210 @@
+## Abstract
+
+Coin Selection algorithms have become a critical part of the Cardano ecosystem. These algorithms choose a set of input Utxos for a given set of new Outputs.
+
+This CIP aims to present optimized coin selection algorithms to enable Wallets and Dapps to interact with smart contracts and native assets such as tokens and NFTs in a more efficient way. The goal of these algorithms is to primarily reduce the size of transactions as well as reduce the mem and step required for smart contract executions.
+
+## Motivation: why is this CIP necessary?
+
+<b>Cardano needs to scale.</b> As the ecosystem grows and new users come in, blocks are regular at 100% load with 88KB per block allocation. Recently, events such as the Lil Sappys NFT mint have pushed the Cardano blockchain load to over 100% for several hours, preventing anyone from using the chain and providing an awful user experience to new users who complain on social media and give the ecosystem a bad reputation.
+
+<b>This can't continue if Cardano is to succeed.</b>
+
+Currently the existing <a href="https://github.com/cardano-foundation/CIPs/tree/master/CIP-0002">CIP-2</a> for coin selection is not sufficient as it was developed in a time before native assets and smart contracts were live on the Cardano Blockchain. Many existing dapps are using extremely poor coin selection resulting in transactions that are 2,3, sometimes 5 times larger then neccesary while also creating outputs that are difficult to handle in future coin selections.
+
+This CIP aims to provide at least 1 algorithm and open the conversation for multiple algorithms to exponentially improve coin selection which will decrease transaction sizes accross the board and increase Cardano's throughput.
+
+Optimizations are priorized in this order for this CIP:
+
+-   Transaction size reduction.
+-   Change Output size reduction for future transactions.
+-   Dust manage to prevent large small dust Utxos from piling up.
+
+## Specification
+
+### Scope
+
+<a href="https://github.com/cardano-foundation/CIPs/tree/master/CIP-0002">CIP-2</a> explains the basics of coin selection, so I won't repeat those here. Instead I will simply present an algorithm for optimizing coin selection. This algorithm will include optimizing the standard coin selection with change selection while also containing suggestions for optimizing collateral selection and optimizing smart contract transactions.
+
+This algorithm will be presented with the following composable structure, allowing for coin and change selection strategies to be composed together:
+
+-   Aggregator Function
+-   Input Selection Function
+-   Change Selection Function
+
+### Algorithms
+
+Random improve was an original algorithm presented in <a href="https://github.com/cardano-foundation/CIPs/tree/master/CIP-0002">CIP-2</a>, this algorithm had a series of good ideas that can be expanded upon here. The first proposed algorithm will simply be an Optimized Random Improve implementation.
```

Comment:
```suggestion
Random Improve was an original algorithm presented in <a href="https://github.com/cardano-foundation/CIPs/tree/master/CIP-0002">CIP-2</a>, this algorithm had a series of good ideas that can be expanded upon. The proposed algorithm will be an Optimized Random Improve implementation.
```

Reply:
One small point: the original author of this algorithm (Edsko de Vries) called it "Random-Improve" (with a hyphen).

I would propose to use the same style in this document, if possible.

Source: [Self Organisation in Coin Selection](https://iohk.io/en/blog/posts/2018/07/03/self-organisation-in-coin-selection/).

Reply:
@NicholasMaselli - unless you can demonstrate that your algorithm is an optimisation of some other "Random Improve" algorithm besides the one @jonathanknowles refers to, editorially we would insist upon changing this term everywhere.

Also: a link to this paper, e.g. as a reference implementation, should be added to the mandatory but currently non-existent _Rationale_ section (see https://github.com/cardano-foundation/CIPs/pull/785#discussion_r1527167129)  Again, if it's a derivation of that algorithm, it's too integral to the CIP itself to be simply included in a _References_ section at the end.

Reply:
The Random-Improve paper (Self Organisation in Coin Selection) is linked/referenced from CIP-2 :)